### PR TITLE
<add>(syncLog): add log when node block number behind of the highest …

### PR DESCRIPTION
**Pull request by LucasLi1024**
…block number

### CODE REVIEW

File: bcos-sync/bcos-sync/BlockSync.h [hunk 1]
```diff
@@ -114,6 +114,7 @@ class BlockSync : public BlockSyncInterface,
     void fetchAndSendBlock(DownloadRequestQueue::Ptr _reqQueue, bcos::crypto::PublicPtr _peer,
         bcos::protocol::BlockNumber _number);
     void printSyncInfo();
+    void checkSyncBlockNumber();
```
Comment (JimmyShi22) [inline bcos-sync/bcos-sync/BlockSync.h:117]: printBehindPeers()

File: bcos-sync/bcos-sync/BlockSync.cpp [hunk 2]
```diff
@@ -180,6 +181,30 @@ void BlockSync::printSyncInfo()
                        << "            --------------------------------------------";
 }
 
+void BlockSync::checkSyncBlockNumber()
+{
+    if (utcTime() - m_lastLogUtcTime <= m_blockNumLogInterval)
+    {
+        return;
+    }
```
Comment (JimmyShi22) [inline bcos-sync/bcos-sync/BlockSync.cpp:189]:
```suggestion
    static auto astLogUtcTime = utcTime();
    if (utcTime() - lastLogUtcTime <= m_blockNumLogInterval)
    {
        return;
    }
```

File: bcos-sync/bcos-sync/BlockSync.cpp [hunk 3]
```diff
@@ -180,6 +181,30 @@ void BlockSync::printSyncInfo()
...
+            BLKSYNC_LOG(WARNING) << "The lowest blocknumber is too far behind the highest block "
+                                    "number queried by getSyncStatus, "
+                                 << "[highest block number: " << knownHighestNumber
+                                 << ", current nodeid: " << _p->nodeId()
+                                 << ", current block number: " << _p->number() << "]";
+        }
+        return true;
+    });
+
+    m_lastLogUtcTime = utcTime();
```
Comment (JimmyShi22) [inline bcos-sync/bcos-sync/BlockSync.cpp:205]:
```suggestion
    lastLogUtcTime = utcTime();
```

File: bcos-sync/bcos-sync/BlockSync.cpp [hunk 4]
```diff
@@ -36,6 +36,7 @@ BlockSync::BlockSync(BlockSyncConfig::Ptr _config, unsigned _idleWaitMs)
     m_syncStatus(std::make_shared<SyncPeerStatus>(_config)),
     m_downloadingQueue(std::make_shared<DownloadingQueue>(_config))
 {
+    m_lastLogUtcTime = utcTime();
```
Comment (JimmyShi22) [inline bcos-sync/bcos-sync/BlockSync.cpp:39]:
```suggestion
```

File: bcos-sync/bcos-sync/BlockSync.h (hunk 1 above)
Comment (LucasLi1024) [reply bcos-sync/bcos-sync/BlockSync.h]: fixed

File: bcos-sync/bcos-sync/BlockSync.cpp (hunk 2 above)
Comment (LucasLi1024) [reply bcos-sync/bcos-sync/BlockSync.cpp]: fixed

File: bcos-sync/bcos-sync/BlockSync.cpp (hunk 3 above)
Comment (LucasLi1024) [reply bcos-sync/bcos-sync/BlockSync.cpp]: fixed

File: bcos-sync/bcos-sync/BlockSync.cpp (hunk 4 above)
Comment (LucasLi1024) [reply bcos-sync/bcos-sync/BlockSync.cpp]: fixed

**Review by JimmyShi22 — APPROVED** [review]

**Review by bxq2011hust — DISMISSED** [review]

File: bcos-sync/bcos-sync/BlockSync.cpp [hunk 5]
```diff
@@ -180,6 +180,31 @@ void BlockSync::printSyncInfo()
...
+    static auto lastLogUtcTime = utcTime();
+    if (utcTime() - lastLogUtcTime <= m_blockNumLogInterval)
+    {
+        return;
+    }
+
+    auto knownHighestNumber = m_config->knownHighestNumber();
+    auto blockNumThreshold = m_blockNumThreshold;
+    m_syncStatus->foreachPeer([knownHighestNumber, blockNumThreshold](PeerStatus::Ptr _p) {
+        if (std::abs(_p->number() - knownHighestNumber) > blockNumThreshold)
```
Comment (kyonRay) [inline bcos-sync/bcos-sync/BlockSync.cpp:194]: add if node is sealer, observer not print error log

Comment (LucasLi1024) [reply bcos-sync/bcos-sync/BlockSync.cpp]: fixed

**Review by JimmyShi22 — APPROVED** [review]

**Pull request merged**